Tests/tests.py

```diff
@@ -1,118 +1,239 @@
+            self.assertEqual(response.status_code, 401)
 
     def tearDown(self):
-        app.db.session.remove()
+        db.session.remove()
 
 
-class TestBucketList(unittest.TestCase):
-    def setUp(self):
-        app.flask_app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///' + BASE_DIR + '/bucket_list_test.db'
-        app.db.session.close()
-        app.db.create_all()
-        self.client = app.flask_app.test_client()
-        # create a user
-        payload = {"first_name": "Sharon", "last_name": "Njihia",
-                   "email": "sharonkaren@gmail.com", "password": "password"}
-        response = self.client.post("/v1/auth/register", data=json.dumps(payload),
-                                    content_type='application/json')
-        print(response.status_code)
-
+class TestBucketList(BaseTestCase):
     def test_if_user_can_create_bucket_list(self):
         """ Test if a user can create a bucket list"""
-
-        # generate a token for adding the data
-        user_credentials = {"email": "sharonkaren@gmail.com", "password": "password"}
-        auth_token = self.client.post("/auth", data=json.dumps(user_credentials),
-                                      content_type='application/json')
-        print(auth_token.data)
-
         # create a bucket list
-        payload = {"bucket_list_id": 89, "bucket_list_name": "sharon", "created_by": 10012}
-        response = self.client.post("/v1/bucketlists", data=json.dumps(payload),
-                                    content_type='application/json')
-        self.assertEqual(response.status_code, 201)
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
+        with flask_app.app_context():
+            payload = {"bucket_list_id": 89, "bucket_list_name": "sharon", "created_by": 10012}
+            response = self.client.post(url_for('get_and_post_bucket_list'), data=json.dumps(payload),
+                                        headers={'Content-Type': 'application/json', 'Authorization': self.token})
+            self.assertEqual(response.status_code, 201)
+
+    def test_if_user_can_get_bucket_lists(self):
+        """ Test if a user can get bucket lists """
+        with flask_app.app_context():
+            response = self.client.get(url_for('get_and_post_bucket_list', limit=1, page=1),
+                                       headers={'Content-Type': 'application/json', 'Authorization': self.token})
+            self.assertEqual(response.status_code, 200)
+
+    def test_if_the_user_can_filter_bucket_lists(self):
+        """ Test if the user can filter the bucket list results """
+        with flask_app.app_context():
+            response = self.client.get(url_for('get_and_post_bucket_list', limit=1, page=1, q='xys'),
+                                       headers={'Content-Type': 'application/json', 'Authorization': self.token})
+            self.assertEqual(response.status_code, 200)
+
+
+class TestSingleBucketList(BaseTestCase):
```

Comment:
For most of these tests checking the response status code only might not be enough. You can consider checking the count in the db for posts & deletes and the data in the response for gets